kernel.spec.in

```diff
@@ -402,9 +402,17 @@ done
 %posttrans
 /bin/kernel-install add %{kernelrelease} /boot/vmlinuz-%{kernelrelease} || exit $?
 
+# with kernel-4.14+ plymouth detects hvc0 serial console and forces text boot
+# we simply make plymouth ignore it to recover the splash screen
+if [ -f /etc/default/grub ]; then
+    if ! grep -q plymouth.ignore-serial-consoles /etc/default/grub; then
+        echo 'GRUB_CMDLINE_LINUX="$GRUB_CMDLINE_LINUX plymouth.ignore-serial-consoles"' >> /etc/default/grub
+    fi
+fi
```

Comment:
This option should be added before call to `kernel-install` (which, among other things, will regenerate grub config). Otherwise it won't be included in grub.cfg, until manual `grub2-mkconfig` call (or next update).

Also, similar thing needs to be done for UEFI - `/boot/efi/EFI/qubes/xen.cfg`. There, you need to append it to `kernel=` line.

Reply:
Yes I did it manually when testing. For xen.cfg, it has to be (only?) added in anaconda. I will do the commit soon.

Reply:
No, xen.cfg should also be adjusted on update. See xen.spec, there is similar thing for xen options.